packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -135,6 +143,14 @@ export interface TableOptions {
    * @experimental
    */
   readonly replicationRegions?: string[];
+
+  /**
+   * External KMS key to use for table encryption.
+   *
+   * @default - If encryption is set to "Customer managed" and this property is undefined,
+   * a new KMS key will be created and associated with this table.
+   */
+  readonly encryptionKey?: kms.IKey;
```

Comment:
One thing worth nothing - from the [DynamoDB documentation](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/encryption.usagenotes.html):

> You cannot use a customer managed CMK to encrypt global tables. For more information, see Global Tables: Multi-Region Replication with DynamoDB.

It's also worth checking for at construction (i.e: `replicationRegions` and `encryptionKey`/`TableEncryption.CUSTOMER_MANAGED` cannot be both set).

Reply:
Yes, good point. I have added this check in `parseEncryption` function. 

I also removed `arnForObjects` function as objects are not applicable in the context of DynamoDB table. For granting permission to stream / index / ... ARNs, we have the following condition in `grant` function:
```ts
      resourceArns: [
        this.tableArn,
        Lazy.stringValue({ produce: () => this.hasIndex ? `${this.tableArn}/index/*` : Aws.NO_VALUE }),
        ...this.regionalArns,
        ...this.regionalArns.map(arn => Lazy.stringValue({
          produce: () => this.hasIndex ? `${arn}/index/*` : Aws.NO_VALUE,
        })),
      ],
``` 
Hence, `arnForObjects` function is not required in this context.